File: ejTasks/start_here/Authorization/AuthMgr.java
package Authorization;

import java.nio.file.Files;
import java.nio.file.Path;
import java.util.HashMap;
import java.util.HashSet;
import java.util.List;
import java.util.Map;
import java.util.Set;
import java.util.logging.Level;
import java.util.logging.Logger;

public class AuthMgr {
	public static final String URGENT_TASK = "urgenttask";
	public static final String VIEW_ACTIVE = "viewactive";

	/*package*/ static final Map<String, String> INVALID_AUTH_EXPLANATIONS = new HashMap<String, String>();
	static {
		INVALID_AUTH_EXPLANATIONS.put(URGENT_TASK, "mark a task as urgent");
		INVALID_AUTH_EXPLANATIONS.put(VIEW_ACTIVE, "view active tasks");
	}

	public static boolean isAllowed(Authorization a_autherization, String a_right) {
	    return a_autherization.allows(a_right);
	}

	private final static Path AUTH_FILENAME = Path.of("auth.txt");
	public static Authorization getAuthorization(String a_username, Logger a_logger) {
	    Set<String> allowed = new HashSet<String>(); // Welcome/Allow list according to Easy to Use Safely
	    try {
            List<String> lines = Files.readAllLines(AUTH_FILENAME);
	        for (String line : lines) {
	            if (line.isBlank()) {
	                continue; // skip empty lines
	            }
	            
	            String[] parts = line.split(":");
	            if (parts.length != 2) {
	                a_logger.log(Level.SEVERE, String.format("Auth line invalid: %s", line));
	                allowed.clear(); // if there's a format error I mistrust EVERYTHING
	                break;
	            }
	            else if (parts[0].equals(a_username)) {
	                allowed.add(parts[1]);
	            }
	        }
	    }
	    catch (Exception ex) {
	        a_logger.log(Level.WARNING, String.format("No auth file %s found -or- empty, or an error happened: %s", AUTH_FILENAME, ex.getMessage()));
	    }
	    return new Authorization(a_username, allowed);
	}
}

File: sharpTasks/start_here/authorization/AuthMgr.cs
using Microsoft.Extensions.Logging;

namespace AuthorizationNS
{
    public static class AuthMgr
    {
        public const string URGENT_TASK = "urgenttask";
        public const string VIEW_ACTIVE = "viewactive";

        internal static readonly Dictionary<string, string> INVALID_AUTH_EXPLANATIONS = new Dictionary<string, string>
        {
            { URGENT_TASK, "mark a task as urgent" },
            { VIEW_ACTIVE, "view active tasks" }
        };

        private static readonly string AUTH_FILENAME = "auth.txt";

        public static bool IsAllowed(Authorization a_authorization, string a_right)
        {
            return a_authorization.Allows(a_right);
        }

        public static Authorization GetAuthorization(string a_username, ILogger a_logger)
        {
            ISet<string> allowed = new HashSet<string>();
            try
            {
                string[] lines = File.ReadAllLines(AUTH_FILENAME);
                foreach (string a_line in lines)
                {
                    if (string.IsNullOrWhiteSpace(a_line))
                    {
                        continue; // skip empty lines
                    }

                    string[] a_parts = a_line.Split(':');
                    if (a_parts.Length != 2)
                    {
                        a_logger.LogError("Auth line invalid: {line}", a_line);
                        allowed.Clear(); // if there's a format error I mistrust EVERYTHING
                        break;
                    }
                    else if (string.IsNullOrEmpty(a_parts[0]) || a_parts[0] == a_username)
                    {
                        allowed.Add(a_parts[1]);
                    }
                }
            }
            catch (Exception a_ex)
            {
                a_logger.LogWarning("No auth file {filename} found -or- empty, or an error happened: {ex}", AUTH_FILENAME, a_ex.Message);
            }
            return new Authorization(a_username, allowed);
        }
    }
}
